homeassistant/components/tag/__init__.py

```diff
@@ -254,7 +256,9 @@ async def tag_change_listener(
             )
         if change_type == collection.CHANGE_ADDED:
             # When tags are added to storage
-            entity = _create_entry(entity_registry, updated_config[CONF_ID], None)
+            entity = _create_entry(
+                entity_registry, updated_config[CONF_ID], updated_config.get(CONF_NAME)
+            )
```

Comment:
```suggestion
            entity = _create_entry(entity_registry, updated_config[CONF_ID], None)
```
Entity is already created with the correct name so we don't need to provide it again.